File: 백준/Bronze/21918. 전구/전구.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int N = sc.nextInt();
        int M = sc.nextInt();
        int[] bulb = new int[N];

        for (int i = 0; i < N; i++) {
            bulb[i] = sc.nextInt();
        }

        for (int i = 0; i < M; i++) {
            int a = sc.nextInt();
            int b = sc.nextInt() - 1;
            int c = sc.nextInt();

            if (a == 1) {
                bulb[b] = c;
            } else if (a == 2) {
                for (int j = b; j < c; j++) {
                    bulb[j] = (bulb[j] + 1) % 2;
                }
            } else if (a == 3) {
                for (int j = b; j < c; j++) {
                    bulb[j] = 0;
                }
            } else {
                for (int j = b; j < c; j++) {
                    bulb[j] = 1;
                }
            }
        }

        for (int i = 0; i < N; i++) {
            System.out.print(bulb[i] + " ");
        }
        System.out.println();
        sc.close();
    }
}

File: 백준/Bronze/21918. 전구/전구.c
#include <stdio.h>

int main() {
    int N, M;
    scanf("%d %d", &N, &M);
    
    int bulb[N];
    for (int i = 0; i < N; i++) {
        scanf("%d", &bulb[i]);
    }

    for (int i = 0; i < M; i++) {
        int a, b, c;
        scanf("%d %d %d", &a, &b, &c);
        b--;
        if (a == 1) {
            bulb[b] = c;
        } else if (a == 2) {
            for (int j = b; j < c; j++) {
                bulb[j] = (bulb[j] + 1) % 2;
            }
        } else if (a == 3) {
            for (int j = b; j < c; j++) {
                bulb[j] = 0;
            }
        } else {
            for (int j = b; j < c; j++) {
                bulb[j] = 1;
            }
        }
    }

    for (int i = 0; i < N; i++) {
        printf("%d ", bulb[i]);
    }
    printf("\n");
    return 0;
}